handler/src/main/java/io/netty/handler/ssl/OpenSslCachingKeyMaterialProvider.java

```diff
@@ -44,6 +46,10 @@ OpenSslKeyMaterial chooseKeyMaterial(ByteBufAllocator allocator, String alias) t
                 return null;
             }
 
+            if (cache.size() > maxCachedEntries) {
```

Comment:
The actual cache size might larger than maxCachedEntries since this check is not thread-safe, but it seems fine to me. Having LRU caching might be overkilled, probably we can implement that if necessary in the future. 

Can you update the doc of this class about the newly added max cached entries behavior?

Reply:
well it is thread-safe but it is not 100 % accurate :) That said I think it will be good enough and is cheap-ish.